File: CycleChesscpp/app/src/main/java/com/serebryakov/cyclechesscpp/application/model/game/PieceType.kt
package com.serebryakov.cyclechesscpp.application.model.game

enum class PieceType {
    EMPTY,
    PAWN,
    KING,
    ROOK,
    BISHOP,
    QUEEN,
    KNIGHT,
}

fun pieceTypeToString(pieceType: PieceType): String {
    return when(pieceType) {
        PieceType.EMPTY -> "0"
        PieceType.PAWN -> "1"
        PieceType.KING -> "2"
        PieceType.ROOK -> "3"
        PieceType.BISHOP -> "4"
        PieceType.QUEEN -> "5"
        PieceType.KNIGHT -> "6"
    }
}

fun Int.toPieceType(): PieceType {
    return when(this) {
        1 -> PieceType.PAWN
        2 -> PieceType.KING
        3 -> PieceType.ROOK
        4 -> PieceType.BISHOP
        5 -> PieceType.QUEEN
        6 -> PieceType.KNIGHT
        else -> PieceType.EMPTY
    }
}

fun Char.toPieceType(): PieceType {
    return when(this) {
        'b' -> PieceType.BISHOP
        'k' -> PieceType.KING
        'q' -> PieceType.QUEEN
        'r' -> PieceType.ROOK
        'n' -> PieceType.KNIGHT
        'p' -> PieceType.PAWN
        else -> PieceType.EMPTY
    }
}

File: CycleChessBot/bot/chess_game/piece_type.py
from enum import Enum

class PieceType(Enum):
    EMPTY = 0
    PAWN = 1
    KING = 2
    ROOK = 3
    BISHOP = 4
    QUEEN = 5
    KNIGHT = 6

    def __str__(self) -> str:
        return str(self.value)

def get_all_piece_types() -> list[PieceType]:
    return [
        PieceType.BISHOP,
        PieceType.KING,
        PieceType.KNIGHT,
        PieceType.PAWN,
        PieceType.QUEEN,
        PieceType.ROOK,
    ]